plasmapy/particles/tests/test_ionization_state.py

```diff
@@ -1,19 +1,100 @@
-import pytest
 import collections
 
-import numpy as np
 import astropy.units as u
+import numpy as np
+import pytest
 
-from plasmapy.particles.ionization_state import IonizationState
-from plasmapy.utils.pytest_helpers import run_test
-from plasmapy.particles.exceptions import AtomicError, InvalidIsotopeError
 from plasmapy.particles import (
+    Particle,
     atomic_number,
     atomic_symbol,
-    particle_symbol,
     isotope_symbol,
-    Particle,
+    particle_symbol,
 )
+from plasmapy.particles.exceptions import InvalidIsotopeError, ParticleError
+from plasmapy.particles.ionization_state import IonicFraction, IonizationState
+from plasmapy.utils.pytest_helpers import run_test
+
+ionic_fraction_table = [
+    ("Fe 6+", 0.52, 5.2e-6 * u.m ** -3),
+    ("He 1+", None, None),
+    ("H-2 0+", None, None),
+]
+
+
+@pytest.mark.parametrize("ion, ionic_fraction, number_density", ionic_fraction_table)
+def test_ionic_fraction_attributes(ion, ionic_fraction, number_density):
+
+    try:
+        instance = IonicFraction(
+            ion=ion, ionic_fraction=ionic_fraction, number_density=number_density
+        )
+    except Exception:
+        pytest.fail(
+            f"Unable to instantiate IonicFraction for ion = {ion}, "
+            f"ionic_fraction = {ionic_fraction}, and number_density = "
+            f"{number_density}"
+        )
```

Comment:
This could just be the `IonicFraction` call - pytest will definitely return test pytest on a failure. There's no need to write these exceptions.

Reply:
Yeah...the corresponding error message should go in `IonicFraction.__init__`.  That would clean up these and other tests.  I think I put the extra error messages in because the OS on my home computer had just given me a "Oops... Something has gone wrong. Please contact your system administrator." error while booting.